packages/block-editor/src/components/writing-flow/use-tab-nav.js

```diff
@@ -45,7 +45,11 @@ export default function useTabNav() {
 		} else if ( hasMultiSelection() ) {
 			container.current.focus();
 		} else if ( getSelectedBlockClientId() ) {
-			getLastFocus()?.current.focus();
+			container.current
+				.querySelector(
+					`[data-block="${ getSelectedBlockClientId() }"]`
+				)
+				.focus();
```

Comment:
@ellatrix suggested this, so it's good :)

Reply:
We should probably add a non-hook util at some point similar to `useBlockRef`